src/shared/Core/Authentication/MicrosoftAuthentication.cs

```diff
@@ -229,20 +247,36 @@ private async Task<AuthenticationResult> GetAccessTokenSilentlyAsync(IPublicClie
             if (PlatformUtils.IsWindows())
             {
                 // If we have a parent window ID then use that, otherwise use the hosting terminal window.
-                IntPtr parentHandle;
                 if (!string.IsNullOrWhiteSpace(Context.Settings.ParentWindowId) &&
                     int.TryParse(Context.Settings.ParentWindowId, out int hWndInt) && hWndInt > 0)
                 {
-                    parentHandle = new IntPtr(hWndInt);
+                    Context.Trace.WriteLine($"Using provided parent window ID '{hWndInt}' for MSAL authentication dialogs.");
+                    appBuilder.WithParentActivityOrWindow(() => new IntPtr(hWndInt));
                 }
                 else
                 {
                     IntPtr consoleHandle = Kernel32.GetConsoleWindow();
-                    parentHandle = User32.GetAncestor(consoleHandle, GetAncestorFlags.GetRootOwner);
-                }
+                    IntPtr parentHandle = User32.GetAncestor(consoleHandle, GetAncestorFlags.GetRootOwner);
 
-                Context.Trace.WriteLine($"Using parent window ID '{parentHandle}' for MSAL authentication dialogs.");
-                appBuilder.WithParentActivityOrWindow(() => parentHandle);
+                    // If we don't have a console window then create a dummy top-level window (for .NET Framework)
+                    // that we can use as a parent. When not on .NET Framework just use the Desktop window.
+                    if (parentHandle != IntPtr.Zero)
+                    {
+                        Context.Trace.WriteLine($"Using console parent window ID '{parentHandle}' for MSAL authentication dialogs.");
+                        appBuilder.WithParentActivityOrWindow(() => parentHandle);
+                    }
+                    else if (enableBroker) // Only actually need to set a parent window when using the Windows broker
+                    {
+#if NETFRAMEWORK
+                        Context.Trace.WriteLine($"Using dummy parent window for MSAL authentication dialogs.");
+                        _dummyWindow = new DummyWindow();
+                        appBuilder.WithParentActivityOrWindow(_dummyWindow.ShowAndGetHandle);
+#else
+                        Context.Trace.WriteLine($"Using desktop parent window for MSAL authentication dialogs.");
+                        parentHandle = User32.GetDesktopWindow();
+#endif
```

Comment:
Is this alternative needed? Since we only need to do this if we're using the broker and the broker is only available on Windows, wouldn't we always be on .NET Framework?

I know there has been some talk with Bogdan about the macOS broker, though, so if you've added it in anticipation of that, then that's fine!

Reply:
You are correct that in most cases you'd be using .NET Framework on Windows, however if you're using the .NET Tool on Windows then that's using .NET 6... but then again we'd never enable the broker.. so yes.. I'll remove this 😃